docs/protocol/Constants.md

```diff
@@ -0,0 +1,54 @@
+# Constants
+
+### Types
+
+These are the list of types used to represent various properties within the
+protocol.
+
+| Alias                   | Shorthand | Type    |
+|-------------------------|-----------|---------|
+| Amount                  | Amount    | u64     |
+| Curve                   | Curve     | u16     |
+| Coin                    | Coin      | u32     |
+| Global Validator Set ID | GVSID     | u32     |
+| Validator Set Index     | VS        | u8      |
+| Key                     | Key       | Vec<u8> |
+
+### Curves
+
+Integer IDs for various curves. It should be noted some curves may be the same,
+yet have distinct IDs due to having different basepoints, and accordingly
+different keys. For such cases, the processor is expected to create one secret
+per curve, and then use DLEq proofs to port keys to other basepoints as needed.
+
+| Curve     | ID |
+|-----------|----|
+| Secp256k1 | 0  |
+| Ed25519   | 1  |
+
+### Networks
+
+Every network connected to Serai operates over a specific curve. While the
+processor generates keys for curves, these keys are bound to specific networks
+via an additive offset created by hashing the network's name (among other
+things). The network's key is used for all coins on that network.
+
+Networks are not acknowledged by the Serai network, solely by the processor.
```

Comment:
This could be moved under Serai, with the benefit that validator sets would be defined over Networks, not over Coins. This is a level of indirection though, when handling Oraclizations, despite slightly cleaning the validator set definition (and ensuring that property now), So... pros and cons? It may be worth to move it on chain, have that literal impl, it may be worth it to leave it off chain, as it has no direct reason to be on?